public/components/visualizations/charts/lines/line.tsx

```diff
@@ -81,6 +83,32 @@ export const Line = ({ visualizations, layout, config }: any) => {
       colorTheme.find((colorSelected) => colorSelected.name.name === field)?.color) ||
     PLOTLY_COLOR[index % PLOTLY_COLOR.length];
 
+  const checkIfMetrics = (metricsSchema: any) => {
+    if (isEmpty(metricsSchema)) return false;
+
+    if (
+      metricsSchema.length === 3 &&
+      metricsSchema.every((schemaField) =>
+        ['@labels', '@value', '@timestamp'].includes(schemaField.name)
+      )
+    ) {
+      return true;
+    }
+    return false;
+  };
+
+  const preprocessMetricsJsonData = (json): any => {
```

Comment:
please add types for all these any, it's going to be hard later for people to figure out the shape of your params and data along the way

Reply:
In this case, I'm thinking about that this function expects the json data passed in as an array with record to only contain 3 fields and returns an metric array for metrics to be something like

```
interface MetricRecord {
    '@labels': string;
    '@timestamp': string;
    '@value': string;
}

and

interface MetricSeries {
    '@labels': [whatever parsed type we define];
    '@timestamp': number;
    '@value': number;
}

const preprocessMetricsJsonData = (json: MetricRecord[]): MetricSeries[] => {
    ...
}

```

Without this kind of typing in place, in the future we may face issues like
- the caller of the function won't know the structure of the returned data. This could lead to potential runtime errors if the calling function expects a certain shape or property in the returned object, like later in line 193 - 195 the code is expecting those 3 fields in each trace
- Without looking into the function body, and especially when we add more code logics to it later, it's hard to know what the function does and what it returns. This may add difficulties to later function reference or refactoring.

one other thing partially related to this is
- The code is using JSON.parse where the function assumes that the fields @labels, @timestamp, and @value are always strings containing valid JSON. If any of these fields is somehow not passed in as string JSON, the JSON.parse will throw a runtime error that leads to page/functionalities crash which should be captured in compiling stage. (here I think we may also need error handling)

Reply:
Yes.  We'll need to re-think this at a higher level - such as #932 .   Compile-time validations (typesript types, etc) are unable to validate dynamic run-time code.  The validation must be done in some other fashion.